sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -42,19 +42,20 @@ class CastExprMeta[INPUT <: CastBase](
   val toType = cast.dataType
   var legacyCastToString = ShimLoader.getSparkShims.getLegacyComplexTypeToString()
 
-  override def tagExprForGpu(): Unit = {
-    recursiveTagExprForGpuCheck(fromType)
-  }
+  override def tagExprForGpu(): Unit = recursiveTagExprForGpuCheck()
 
-  def recursiveTagExprForGpuCheck(fromDataType: DataType) {
-    if (!conf.isCastFloatToDecimalEnabled && toType.isInstanceOf[DecimalType] &&
+  private def recursiveTagExprForGpuCheck(
+     fromDataType: DataType = fromType,
+     toDataType: DataType = toType
+  ) {
```

Comment:
Nit: There's some odd 3-space continuation indenting and brace placement here

Reply:
need to tweak intellij code style